lib/config/fileconf.go

```diff
@@ -438,9 +442,15 @@ func (conf *FileConfig) CheckAndSetDefaults() error {
 		}
 	}
 
-	matchers := make([]AWSEC2Matcher, 0, len(conf.SSH.AWSMatchers))
+	matchers := make([]AWSEC2Matcher, 0, len(conf.Discovery.AWSMatchers))
+
+	for _, matcher := range conf.Discovery.AWSMatchers {
+		for _, serviceType := range matcher.Matcher.Types {
+			if serviceType != "ec2" {
```

Comment:
Nit: Can we move "ec2" into a constant and have a global variable somewhere e.g. `supportedDiscoveryServices` that will make it easier to extend this?

And also tweak the message below e.g. "discovery service type does not support %q, supported resource types are: %v" like we do elsewhere.